src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs

```diff
@@ -168,6 +175,19 @@ public override bool Execute()
                 // Summary
                 RestoreSummary.Log(log, restoreSummaries);
 
+                var projectsWithImportsChanged = new StringBuilder();
+                foreach (var summary in restoreSummaries)
+                {
+                    // Success does not matter here, as long as the files are not changed
+                    if(summary.BuildFilesChanged)
```

Comment:
will there be a case where restore failed but still changed the targets/props file?

Reply:
@jainaashish 
Yeah, restoreSuccess is part of the props.